gms/gossiper.cc

```diff
@@ -2217,12 +2217,7 @@ future<> gossiper::add_local_application_state(application_state_map states) {
                 co_await coroutine::return_exception(std::runtime_error(err));
```

Comment:
Isn't `on_internal_error` more appropriate though?

Reply:
Tried it. it breaks scylla
We need to address this somehow as a follow-up.
Maybe even construct the local endpoint_state on demand rather than returning an error so it can keep the application state added in this call (with zero generation).

Reply:
Bottom line is that I don't want to change the behavior in this series.

Reply:
>  Maybe even construct the local endpoint_state on demand rather than returning an error so it can keep the application state added in this call (with zero generation).

We should understand the problem first, it may be something serious.

Reply:
We have a layering problem where raft topology changes wants to update the (local) endpoint state too early, before start_gossping is called, setting status to be normal IIRC.
I think that with consistent topology changes, the endpoint states should be owned and managed by the raft topology layer rather than the gossiper. group0 can also determine the liveness status of nodes.  In this model the gossiper would call raft topology to update the endpoint state with non-topology application states (e.g. RPC_READY), not the other way around.